loggercheck.go

```diff
@@ -25,14 +34,39 @@ func NewAnalyzer() *analysis.Analyzer {
 		Requires: []*analysis.Analyzer{inspect.Analyzer},
 	}
 
-	checkerKeys := strings.Join(loggerCheckersByName.Keys(), ",")
-	a.Flags.Init("loggercheck", flag.ExitOnError)
-	a.Flags.Var(&l.disable, "disable", fmt.Sprintf("comma-separated list of disabled logger checker (%s)", checkerKeys))
+	if !l.disableFlags {
+		defaultFlags(&a.Flags, l)
+	}
+
 	return a
 }
 
+func defaultFlags(fs *flag.FlagSet, l *loggercheck) {
+	checkerKeys := strings.Join(loggerCheckersByName.Keys(), ",")
+	fs.Init("loggercheck", flag.ExitOnError)
+	fs.Var(&l.disable, "disable", fmt.Sprintf("comma-separated list of disabled logger checker (%s)", checkerKeys))
+
+	fs.Func("logger", "add extra logger checker, format: name:packageImport:funcs, "+
+		"example: mylogger:example.com/mylogger:(example.com/mylogger.Logger).Info,(example.com/mylogger.Logger).Error",
+		func(s string) error {
+			if strings.Count(s, ":") != 2 {
+				return errors.New("invalid logger checker")
+			}
+			checker := strings.SplitN(s, ":", 3)
+			for _, v := range checker {
+				if v == "" {
+					return errors.New("invalid logger checker")
+				}
+			}
+			funcNames := strings.Split(checker[2], ",")
+			addLogger(checker[0], checker[1], funcNames)
+			return nil
+		})
+}
+
 type loggercheck struct {
-	disable loggerCheckersFlag // flag -disable
+	disableFlags bool
```

Comment:
I think the `disableFlags` should be removed? Since it's only used during tests, do you mean there is race if we do not use it?

Reply:
it first intorduced because the `disableAll` flag.
since the flags parse is done in the `analysistest.Run`, and it will parse the flags, which cause the opion got overrided.
so this allow us to skip the flags parse/

Reply:
the use case is for golangci-lint like `loggercheck.NewAnalyzer(options...)`

this can be removed now.

for now, we do not have a boolean flag, so this is not affected.
but if we have a boolean flag, this will cause problem